lib/src/Uncertainty/Bayesian/GaussianNonLinearCalibration.cxx

```diff
@@ -364,10 +364,63 @@ void GaussianNonLinearCalibration::run()
     parameterPosterior = KernelSmoothing().build(thetaSample);
   }
   else
-  {
-    GaussianLinearCalibration algo(model_, inputObservations_, outputObservations_, thetaStar, getParameterPrior().getCovariance(), error.getCovariance());
-    algo.run();
-    parameterPosterior = algo.getResult().getParameterPosterior();
+    {
+      // Compute the linearization
+      Function parametrizedModel(model_);
+      parametrizedModel.setParameter(thetaStar);
+      // Flatten everything related to the model evaluations over the input observations
```

Comment:
Is this a copy of this code from lines 105 to 173?https://github.com/openturns/openturns/blob/07a9b2174d4cfbf8382df26ed663e3402558299e/lib/src/Uncertainty/Bayesian/GaussianLinearCalibration.cxx#L105
If yes, then it is mathematically correct, but the code can be factored. Otherwise I do not exactly understand what is done.

Reply:
yes it is. For now, it is more a quick and dirty fix, only to check if it is what we expect from a mathematical point of view and if the implementation is mathematically correct. I postpone the refactoring of the code to after our next technical committee. The idea is to factor things between GaussianNonLinearCalibration and GaussianLinearCalibration the way I did it between NonLinearLeastSquaresCalibration and LinearLestSquaresCalibration, but as it is more involved I prefer to discuss it before to go for a mess.